lib/db/api.js

```diff
@@ -2471,20 +2471,28 @@ export const loadBookableDevicesWithFiltersLock = function(groups, abi, model, t
                     ]
                 }
             )
-            if (limit) {
-                result = result.limit(limit)
-            }
```

Comment:
This will return every document from the database, which can be hefty.  
Maybe use https://www.mongodb.com/docs/manual/reference/operator/aggregation/sample/?